src/julia.h

```diff
@@ -274,13 +283,18 @@ typedef struct _jl_datatype_t {
     uint8_t pointerfree;
     int32_t ninitialized;
     // hidden fields:
+    uint8_t fielddesc_type : 2; // 0 -> 8, 1 -> 16, 2 -> 32
```

Comment:
maybe steal a few bits from the `alignment` field (which should never need to be larger than a byte) instead of adding another 32-bit word here?